### ISSUE TITLE
Replace F.dropout to nn.Dropout() at 2_classify_names_with_MLP.ipynb

### ISSUE THREAD

**Issue by renatoviolin** [joosthub/pytorch-nlp-tutorial-sj2019#1]
I've been having random predictions given the same inputs, even tough using the eval() method.

I realized the problem was the use of dropout. To eval() disable the dropout during inference time it is necessary to use the nn.Dropout module instead of functional F.dropout().

So, the code would be:

```
class MLP(nn.Module):
    def __init__(self, vocab_size, embedding_dim, hidden_dim, output_size, maxlen, dropout_p=0.5):
        super(MLP, self).__init__()
        self.embeddings = nn.Embedding(vocab_size, embedding_dim, padding_idx=0)
        self.fc1 = nn.Linear(embedding_dim * maxlen, hidden_dim)
        self.fc2 = nn.Linear(hidden_dim, output_size)
        self.dropout = nn.Dropout(dropout_p)

    def forward(self, x, apply_softmax=False):
        x = self.embeddings(x).view(-1, (embedding_dim * maxlen))
        x = F.relu(self.fc1(x))
        x = self.dropout(x)
        x = self.fc2(x)
        if apply_softmax:
            x = F.softmax(x, dim=1)
        return x
```
